### PR TITLE
Fixes CodeClimate issues subset

### PR DESCRIPTION
**Pull request by dfar-io**
This fixes a subset of CodeClimate issues in the Freshli library. The resulting files changed can be found [here].(https://codeclimate.com/github/corgibytes/freshli/pull/222)

In addition, I've:
* Added documentation on how to run CodeClimate locally.
* Added a `.codeclimate.json` file that will allow for the same checks to be run both locally and remotely.

### CODE REVIEW

**Review by dan-hein — CHANGES_REQUESTED** [review]
These are some really good changes! There are a few minor adjustments that we discussed that would be good to change for readability, but otherwise, this is a great step forward! The main thing I'd focus on next is making sure that we have proper test coverage for the newly broken out methods.

File: Freshli/Languages/Python/PythonVersionInfo.cs [hunk 1]
```diff
@@ -73,11 +72,9 @@ public enum SuffixType {
       @"(\.(dev)([0-9]+))?(?-i)$"
     );
 
-    public PythonVersionInfo() { }
-
-    public PythonVersionInfo(string version, DateTime datePublished) {
+    public PythonVersionInfo(string version, DateTime? datePublished = null) {
       Version = version;
-      DatePublished = datePublished;
+      if (datePublished.HasValue) { DatePublished = datePublished.Value; }
```
[the remark this answers is not in the document]
Comment (dfar-io) [reply Freshli/Languages/Python/PythonVersionInfo.cs:77]: Looks like without changing `IVersionInfo`, I need to not assign to `DatePublished` unless there is a value provided... leaving this as is.

File: Freshli/LibYearCalculator.cs [hunk 2]
```diff
@@ -23,113 +23,137 @@ IManifest manifest
...
+            $"setting value to 0: {packageResult}");
         }
       }
 
-      return result;
+      _logger.Trace($"PackageResult: {packageResult.ToString()}");
+      return packageResult;
     }
 
+    // We should look into moving this into LibYearPackageResult.cs
```
[the remark this answers is not in the document]
Comment (dfar-io) [reply Freshli/LibYearCalculator.cs:102]: On second thought, this is going to take more work than I would like to tackle, going to leave as is.

File: Freshli/Languages/Python/PythonVersionInfo.cs [hunk 3]
```diff
@@ -219,52 +202,44 @@ PythonVersionInfo otherVersionInfo
     private int ComparePostReleaseIncrementIfPostSuffix(
       PythonVersionInfo otherVersionInfo
     ) {
```
[the remark this answers is not in the document]
Comment (dfar-io) [reply Freshli/Languages/Python/PythonVersionInfo.cs:204]: I remember now - ends up changing a number of files that are using the cast for comparison - I think we should revisit this if it becomes a problem.

**Review by dan-hein — APPROVED** [review]

**Pull request merged**